crates/client/src/test_generator.rs

```diff
@@ -603,102 +577,63 @@ impl TestGenerator {
+                    )
+                    .parse()
+                    .unwrap()
+                };
+                if let toml::Value::Table(table) = dep {
                     let (name, value) = table.into_iter().next().unwrap();
-                    dev_deps.entry(name).or_insert(value);
+                    client_toml_deps.entry(name).or_insert(value.clone());
                 }
             }
-            fs::write(&cargo_toml_path, content.to_string()).await?;
+            fs::write(cargo_path, cargo_toml_content.to_string()).await?;
         } else {
-            println!("Skipping adding programs to Cargo.toml ...");
-        }
-    }
-
-    /// Scans `programs` directory and returns a list of `toml::Value` programs and their paths.
-    async fn get_programs(&self, cargo_dir: &PathBuf) -> Result<Vec<Value>, Error> {
-        let programs = self.root.join("programs");
-        if !programs.exists() {
-            println!("Programs folder does not exist.");
-            return Ok(Vec::new());
-        }
-        println!("Searching for programs ...");
-        let mut program_names: Vec<Value> = vec![];
-        let programs = std::fs::read_dir(programs)?;
-        for program in programs {
-            let file = program?;
-            if file.path().is_dir() {
-                let path = file.path().join(CARGO_TOML);
-                if path.exists() {
-                    let dependency = self.get_program_dep(&path, cargo_dir).await?;
-                    program_names.push(dependency);
-                }
-            }
+            throw!(Error::NoProgramsFound)
         }
-        Ok(program_names)
     }
+}
 
-    /// Scans `programs` directory and returns a list of names of libraries
-    async fn get_program_lib_names(&self) -> Result<Vec<String>, Error> {
-        let programs = self.root.join("programs");
-        if !programs.exists() {
-            println!("Programs folder does not exist.");
-            return Ok(Vec::new());
-        }
-        println!("Searching for programs ...");
-        let mut program_names: Vec<String> = vec![];
-        let programs = std::fs::read_dir(programs)?;
-        for program in programs {
-            let file = program?;
-            if file.path().is_dir() {
-                let path = file.path().join(CARGO_TOML);
-                if path.exists() {
-                    let content: Value = fs::read_to_string(&path).await?.parse()?;
-                    let name = content
-                        .get("lib")
-                        .and_then(Value::as_table)
-                        .and_then(|table| table.get("name"))
-                        .and_then(Value::as_str)
-                        .ok_or(Error::CannotParseCargoToml)?;
-                    program_names.push(name.to_string());
-                }
-            }
-        }
-        Ok(program_names)
-    }
+pub fn load_template(file_path: &str) -> Result<String, std::io::Error> {
```

Comment:
The function is ok, I meant the way you do the same thing, because `include_str!` directly includes the contents of the file at compile time. This means there is no runtime file I/O involved...

Reply:
It would also make sense to make the function inline...